boxoffice/models/line_item.py

```diff
@@ -255,12 +257,36 @@ def sales_by_date(dates, user_tz, item_ids):
     return date_sales
 
 
-def sales_delta(user_tz, item_ids):
+def calculate_weekly_sales(item_collection_ids, user_tz, year):
     """
-    Calculates the percentage difference in sales between today and yesterday
+    Calculates sales per week for items in the given set of item_collection_ids in a given year,
+    in the user's timezone.
     """
+    week_sales = db.session.query('sales_week', 'sum').from_statement(text('''SELECT DATE_TRUNC('week', ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone)
+        AS sales_week, SUM(final_amount)
+        FROM line_item INNER JOIN item on line_item.item_id = item.id
+        WHERE status=:status AND item_collection_id IN :item_collection_ids
+        AND ordered_at >= :start_at AND ordered_at < :end_at
+        GROUP BY sales_week ORDER BY sales_week;
+        ''')).params(timezone=user_tz, status=LINE_ITEM_STATUS.CONFIRMED,
+        start_at='{year}-01-01'.format(year=year),
+        end_at='{year}-01-01'.format(year=year+1),
+        item_collection_ids=tuple(item_collection_ids), year='{year}-01-01'.format(year=year)).all()
+    week_sales_dict = {}
```

Comment:
Since there's no order in JSON, `jsonify` doesn't preserve order [1] even if I use `ordereddict`. I'll have to jsonify `ordereddict.items()` which will render a nested list of lists eg: [[1,100.0], [2, 200.0]]. Is that okay, or is the current structure fine?

[1] https://github.com/pallets/flask/issues/974